### PR TITLE
Grouping elements in navbar

### LINKED ISSUE #1060: Link core-packages page
Following #1026. We are missing a link to it.

The page is already online http://babeljs.io/docs/core-packages/.

We could add it in the docs dropdown?

screenshot from 2016-12-08 09-08-27

Or we could replace this page http://babeljs.io/docs/usage/api/ which has the same in it.

@hzoo @existentialism Any ideas?

### PR DESCRIPTION
**Pull request by xtuc**
Fixes #1060

#### Desktop:
nav-desktop

 #### Collasped:
nav-mobile

Given #1042 this is a quick fix.

Navigation data has a bit changed:
* actual item
* subnav
    * name (optionnaly set a group name)
    * items
        * actual item

### CODE REVIEW

File: _data/navigation.yml [hunk 1]
```diff
@@ -3,24 +3,31 @@
...
+        - title: Setup
+          url: /docs/setup/
+        - title: Editors
+          url: /docs/editors
+        - title: Caveats
+          url: /docs/usage/caveats/
+    - name: Packages
+      items:
+        - title: API
+          url: /docs/core-packages/
```
Comment (xtuc) [inline _data/navigation.yml]: `API` is pointing babel's core packages. We can rename it if you want.

**Comment by existentialism** [comment]
As an alternative to the indention, what are your thoughts about:

image

Also, I think renaming "API" -> "Core" under Packages is a little clearer? @hzoo?

**Comment by xtuc** [comment]
Yes, Looks better.
Have you tried with the collasped version ?

**Comment by existentialism** [comment]
Yeah, we can do something similar for now... when we attack the sidebar we may have to rethink a bunch of this anyway:

image

**Comment by xtuc** [comment]
It seems great. Would you push it there so you can merge this.

**Comment by existentialism** [comment]
@xtuc pushed changes, check em out

**Comment by xtuc** [comment]
Nice job, thanks!

LGTM, can we merge this ?

**Comment by hzoo** [comment]
Awesome looks good.

Although now I think maybe would could rename a few things?

Should core be babel-core and polyfill babel-polyfill or is the shorthand better?

Maybe should remove the browser in usage since it redirects to setup anyway?

**Comment by existentialism** [comment]
Good point re: browsers, agree we should drop.

And yeah, while I think Core and Polyfill _look_ better, it might be better to go with babel-core and babel-polyfill to eliminate any chance someone might be confused (esp since babel-register is below, and Register would be pretty strange by itself).

I'll update!

**Pull request merged**